fault-proof/tests/common/env.rs

```diff
@@ -263,50 +281,49 @@ impl TestEnvironment {
     pub fn factory(
         &self,
     ) -> Result<DisputeGameFactoryInstance<impl alloy_provider::Provider + Clone>> {
-        let provider_with_signer = self.provider_with_signer()?;
-        let factory = DisputeGameFactory::new(self.deployed.factory, provider_with_signer.clone());
+        let provider = self.provider_with_role(Role::Proposer)?;
+        let factory = DisputeGameFactory::new(self.deployed.factory, provider);
         Ok(factory)
     }
 
     pub async fn anchor_registry_address(&self, game_address: Address) -> Result<Address> {
-        let provider_with_signer = self.provider_with_signer()?;
-        let anchor_registry_addr =
-            OPSuccinctFaultDisputeGame::new(game_address, provider_with_signer.clone())
-                .anchorStateRegistry()
-                .call()
-                .await?;
+        let provider = self.provider_with_role(Role::Proposer)?;
+        let anchor_registry_addr = OPSuccinctFaultDisputeGame::new(game_address, provider)
+            .anchorStateRegistry()
+            .call()
+            .await?;
         Ok(anchor_registry_addr)
     }
 
     pub async fn anchor_registry(
         &self,
         game_address: Address,
     ) -> Result<AnchorStateRegistryInstance<impl alloy_provider::Provider + Clone>> {
-        let provider_with_signer = self.provider_with_signer()?;
+        let provider = self.provider_with_role(Role::Proposer)?;
         let anchor_registry_addr = self.anchor_registry_address(game_address).await?;
-        let anchor_registry =
-            AnchorStateRegistry::new(anchor_registry_addr, provider_with_signer.clone());
+        let anchor_registry = AnchorStateRegistry::new(anchor_registry_addr, provider);
         Ok(anchor_registry)
     }
 
     pub async fn mock_optimism_portal2(
         &self,
         anchor_registry: AnchorStateRegistryInstance<impl alloy_provider::Provider + Clone>,
     ) -> Result<MockOptimismPortal2Instance<impl alloy_provider::Provider + Clone>> {
-        let provider_with_signer = self.provider_with_signer()?;
+        let provider = self.provider_with_role(Role::Proposer)?;
         let portal_addr = anchor_registry.portal().call().await?;
-        let portal = MockOptimismPortal2::new(portal_addr, provider_with_signer.clone());
+        let portal = MockOptimismPortal2::new(portal_addr, provider);
         Ok(portal)
     }
 
-    pub async fn fault_dispute_game(
+    /// Create a fault dispute game instance with a provider for the specified role
+    pub async fn fault_dispute_game_with_role(
```

Comment:
Quick suggestion: let’s add a `fault_dispute_game()` that defaults to `Role::Proposer`, so we don’t have to pass the role everywhere. A bunch of these methods are always meant to be called by the proposer anyway.